lib/topological_inventory/ansible_tower/receptor/collector.rb

```diff
@@ -35,20 +37,48 @@ def async_collecting_finished(entity_type, refresh_state_uuid, total_parts)
       end
 
       def async_save_inventory(refresh_state_uuid, parser)
+        self.last_save_at = Time.now.utc
         refresh_state_part_collected_at = Time.now.utc
         refresh_state_part_uuid = SecureRandom.uuid
         save_inventory(parser.collections.values, inventory_name, schema_name, refresh_state_uuid, refresh_state_part_uuid, refresh_state_part_collected_at)
       end
 
       def async_sweep_inventory(refresh_state_uuid, sweep_scope, total_parts, refresh_state_started_at)
         logger.sweeping(:start, source, sweep_scope, refresh_state_uuid)
+
         sweep_inventory(inventory_name, schema_name, refresh_state_uuid, total_parts, sweep_scope, refresh_state_started_at)
+        entity_types_collected_cnt.increment # Real finish of entity_type's requests
+
         logger.sweeping(:finish, source, sweep_scope, refresh_state_uuid)
       end
 
       private
 
-      attr_accessor :account_number, :receptor_node
+      attr_accessor :account_number, :entity_types_collected_cnt, :last_save_at, :receptor_node
+
+      def start_collector_threads
+        self.last_save_at = nil
+        entity_types_collected_cnt.value = 0
+
+        super
+      end
+
+      def wait_for_collected_data
+        super
+
+        # Wait until all async responses are sent
+        while entity_types_collected_cnt.value < service_catalog_entity_types.size
+          if last_save_at.nil? || (last_save_at < 2.minutes.ago && last_save_at >= 10.minutes.ago)
```

Comment:
I think we should make this value `10.minutes.ago` configurable, that way we can tweak it as time goes on, some towers might not be able to be collected fully within 10 minutes.

Reply:
I agree, but this value means that there was at least one successful data response in last 10 minutes. Then it resets the 10 minutes counter